messaging/src/main/java/org/axonframework/eventhandling/tokenstore/jdbc/JdbcTokenStore.java

```diff
@@ -157,6 +171,40 @@ public boolean requiresExplicitSegmentInitialization() {
         return true;
     }
 
+    @Override
+    public Optional<String> retrieveStorageIdentifier() throws UnableToRetrieveIdentifierException {
+        return Optional.of(loadConfigurationToken()).map(i -> i.get("id"));
+    }
+
+    private ConfigToken loadConfigurationToken() throws UnableToRetrieveIdentifierException {
+        Connection connection = getConnection();
+        TrackingToken token;
+        try {
+            token = executeQuery(connection, c -> select(connection, CONFIG_TOKEN_ID, CONFIG_SEGMENT, false),
+                                 resultSet -> {
+                                     if (resultSet.next()) {
+                                         return readTokenEntry(resultSet).getToken(serializer);
+                                     } else {
+                                         return null;
+                                     }
+                                 }, e -> new UnableToRetrieveIdentifierException("Exception while attempting to retrieve the config token", e));
+            try {
+                if (token == null) {
+                    token = insertTokenEntry(connection, new ConfigToken(Collections.singletonMap("id", UUID.randomUUID().toString())), CONFIG_TOKEN_ID, CONFIG_SEGMENT);
+                }
+            } catch (SQLException e) {
+                throw new UnableToRetrieveIdentifierException("Exception while attempting to initialize the config token. It may have been concurrently initialized.", e);
+            }
+        } finally {
+            closeQuietly(connection);
+        }
+        return (ConfigToken) token;
+    }
+
+    public Serializer serializer() {
```

Comment:
What do we need this method for? If it's here to stay, I think it deserves some Javadoc though.

Reply:
I remember seeing some requests on the mailinglist of people wanting access to the specific elements being used. In this case, I also needed it in a testcase. So I figured: why not just add it here. Will add javadoc.